catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -311,6 +316,44 @@ public DeleteResponse delete(DeleteRequest deleteRequest)
     //
     // Private helper methods
     //
+    private DeleteRequest rewriteQuery(DeleteRequest deleteRequest)
+            throws UnsupportedQueryException {
+        if (Metacard.ID.equals(deleteRequest.getAttributeName())) {
+            return deleteRequest;
+        }
+
+        Filter filter = frameworkProperties.getFilterBuilder()
+                .anyOf(deleteRequest.getAttributeValues()
+                        .stream()
+                        .map(e -> frameworkProperties.getFilterBuilder()
+                                .attribute(deleteRequest.getAttributeName())
+                                .is()
+                                .equalTo()
+                                .text(e.toString()))
+                        .collect(Collectors.toList()));
+        SourceResponse response = catalogSupplier.get().query(new QueryRequestImpl(new QueryImpl(filter), false));
+        Map<Serializable, Metacard> newUpdateList = response.getResults()
+                .stream()
+                .filter(r -> MetacardVersion.isNotVersion(r.getMetacard()))
+                .collect(Collectors.toMap(r -> (Serializable) r.getMetacard().getId(),
+                        Result::getMetacard));
+
+        // confirm every item from original update list is present.
+        if (!newUpdateList.values()
```

Comment:
The comment and the code disagree here. This code verifies that every metacard in `newUpdateList` was in the original delete request. It will return true even if some metacards from the original delete request are not present in `newUpdateList`.